libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -138,20 +140,57 @@ export class ShellOperationRunner implements IOperationRunner {
   }
 
   private async _executeAsync(context: IOperationRunnerContext): Promise<OperationStatus> {
-    // TERMINAL PIPELINE:
-    //
-    //                             +--> quietModeTransform? --> collatedWriter
-    //                             |
-    // normalizeNewlineTransform --1--> stderrLineTransform --2--> removeColorsTransform --> projectLogWritable
-    //                                                        |
-    //                                                        +--> stdioSummarizer
+    // Only open the *.cache.log file(s) if the cache is enabled.
+    const cacheProjectLogWritable: ProjectLogWritable | undefined = this._buildCacheConfiguration
+      ? new ProjectLogWritable(
+          this._rushProject,
+          context.collatedWriter.terminal,
+          `${this._logFilenameIdentifier}.cache`
+        )
+      : undefined;
+
     const projectLogWritable: ProjectLogWritable = new ProjectLogWritable(
       this._rushProject,
       context.collatedWriter.terminal,
       this._logFilenameIdentifier
     );
 
     try {
+      //#region CACHE LOGGING
+      const cacheDiscardTransform: DiscardStdoutTransform = new DiscardStdoutTransform({
```

Comment:
Can we inline this so we don't create the DiscardStdoutTransform only to immediately discard it in the `--verbose` case?